### PR TITLE
cgi: Fix showing query string in Help

### PR DESCRIPTION
**Pull request by zdohnal**
If a string was searched on CUPS Web UI help page, garbage was printed out in search box. It was because text field pointer was freed before and contained garbage - previously it was variable value which was allocated, so the string survived `cgCleanVariables(()`, but the text field is a pointer into form variables which gets cleaned up.

Fix is to use `strdup()` if `cgiGetTextfield()` returns non-NULL pointer. The binary exits shortly after either way, so memory is taken care of by OS.

### CODE REVIEW

File: cgi-bin/help.c [hunk 1]
```diff
@@ -173,7 +173,7 @@ main(int  argc,				/* I - Number of command-line arguments */
   if (cgiGetVariable("CLEAR"))
     cgiSetVariable("QUERY", "");
 
-  query = cgiGetTextfield("QUERY");
+  query = ( query = (char*)cgiGetTextfield("QUERY")) != NULL ? strdup(query) : NULL;
```
Comment (michaelrsweet) [inline cgi-bin/help.c:176]: Ok this code is hard to follow. Try refactoring, something like:

```c
const char *query = NULL; // …

…

if (cgiGetVariable(“CLEAR”))
  cgiSetVariable(“QUERY”, “”);
else if ((query = cgiGetTextField(“QUERY”)) != NULL)
  query = strdup(query);
```

File: cgi-bin/help.c [hunk 2]
```diff
@@ -28,7 +28,7 @@ main(int  argc,				/* I - Number of command-line arguments */
 		*si;			/* Search index */
   help_node_t	*n;			/* Current help node */
   int		i;			/* Looping var */
-  const char	*query;			/* Search query */
+  char		*query = NULL;			/* Search query */
```
Comment (michaelrsweet) [inline cgi-bin/help.c:31]: This can stay const, but do assign to NULL.

**Review by michaelrsweet — CHANGES_REQUESTED** [review]
Obviously we need to do this, but the proposed change is hard to read. See comments.

File: cgi-bin/help.c (hunk 1 above)
Comment (zdohnal) [reply cgi-bin/help.c]: Thanks, updated!

File: cgi-bin/help.c (hunk 2 above)
Comment (zdohnal) [reply cgi-bin/help.c]: Updated, thanks!

**Review by michaelrsweet — APPROVED** [review]
+1
